src/batch/src/executor/update.rs

```diff
@@ -96,7 +103,8 @@ impl UpdateExecutor {
     #[try_stream(boxed, ok = DataChunk, error = RwError)]
     async fn do_execute(mut self: Box<Self>) {
         let data_types = self.child.schema().data_types();
-        let mut builder = DataChunkBuilder::new(data_types, self.chunk_size);
+        let mut builder = DataChunkBuilder::new(data_types.clone(), self.chunk_size);
+        let mut updated_builder = DataChunkBuilder::new(data_types.clone(), self.chunk_size);
```

Comment:
Some ideas: the `Update` with returning should behave like a `Project`, which does not "buffer" or reorganize the chunks with a builder. 🤔 In this sense, should we directly yield every updated chunk in the execution loop?